### ISSUE TITLE
How to register handlers and listeners programatically from a ServiceProvider

### ISSUE THREAD

**Issue by josecelano** [prooph/laravel-package#13]
Hi. I am using this package in a Laravel aplication where we have different modules/packages. Each modules has its own ServiceProvider and its own comands/events/command-handlers/event-listeners.

I would like to register command handlers and event listeners from the ServiceProvider instead of adding them manually in the config file. As you are using only a single config file and we can not merge recursively the module configuration with the app configuration we have to manually add the configuration  to the `prooph` config file every time we install a new module in the application which contains new comand handlers/event listeners.

Basically we want each module to be auto-installable, and register its own handlers and listeners.

**Comment by bweston92** [comment]
The service provider for each package can register it's own handlers. Something like the following. Remember to register a service locator to "App::make()" the handlers.
```
public function register() : void
{
  $this->registerCommandMap($this->app['config']['mypackage::bus.commands'] ?? []);
  $this->registerQueryMap($this->app['config']['mypackage::bus.queries'] ?? []);
  $this->registerEventMap($this->app['config']['mypackage::bus.events'] ?? []);
}

private function registerCommandMap(array $commands) : void
{
  /** @var \Prooph\ServiceBus\Plugin\Router\CommandRouter */
  $router = $this->app[\Prooph\ServiceBus\Plugin\Router\CommandRouter::class];
  foreach ($commands as $messageName => $handler) {
    $router->route($messageName)->to($handler);
  }
}
private function registerQueryMap(array $queries) : void
{
  /** @var \Prooph\ServiceBus\Plugin\Router\QueryRouter */
  $router = $this->app[\Prooph\ServiceBus\Plugin\Router\QueryRouter::class];
  foreach ($queries as $messageName => $handler) {
    $router->route($messageName)->to($handler);
  }
}
private function registerEventMap(array $events) : void
{
  /** @var \Prooph\ServiceBus\Plugin\Router\EventRouter */
  $router = $this->app[\Prooph\ServiceBus\Plugin\Router\EventRouter::class];
  foreach ($events as $eventName => $handlers) {
    $router->route($eventName);
    array_map([$router, 'andTo'], $handlers);
  }
}
```

**Comment by bweston92** [comment]
@sandrokeil can you mark as awaiting for feedback or put any resources you have here so we can close?

**Comment by bweston92** [comment]
Maybe it is time to close this.
